contracts/src/CrossChainApplications/GETTING_STARTED.md

```diff
@@ -0,0 +1,197 @@
+
+Create a state variable of `ITeleporterMessenger` type called `teleporterMessenger`. Then create a constructor for our contract that takes in an address where the Teleporter Messenger would be deployed on this chain, and set our state variable with it.
+
+```solidity
+contract ExampleCrossChainMessenger {
+    ITeleporterMessenger public immutable teleporterMessenger;
+
+    constructor(address teleporterMessengerAddress) {
+        teleporterMessenger = ITeleporterMessenger(teleporterMessengerAddress);
+    }
+}
+```
+
+### Step 3: Send and Receive
+
+Now that `MyExampleCrossChainMessenger` has an instantiation of `ITeleporterMessenger`, the next step is to add in functionality of sending and receiving arbitrary string data between chains.
+
+To start, create the function declarations for `sendMessage`, which will send string data cross-chain to the specified destination address' receiver. This function allows callers to specify the destination chain ID, destination address to send to, relayer fees, required gas limit for message execution on destination address, and the actual message data.
+
+```solidity
+// Send a new message to another chain.
+function sendMessage(
+    bytes32 destinationChainID,
+    address destinationAddress,
+    address feeTokenAddress,
+    uint256 feeAmount,
+    uint256 requiredGasLimit,
+    string calldata message
+) external returns (uint256 messageID) {}
+```
+
+`MyExampleCrossChainMessenger` also needs to implement `ITeleporterReceiver` by adding the method `receiveTeleporterMessage` that receives the cross-chain messages from Teleporter.
+
+```solidity
+// Receive a new message from another chain.
+function receiveTeleporterMessage(
+    bytes32 originChainID,
+    address originSenderAddress,
+    bytes calldata message
+) external {}
+```
+
+Now it's time to implement the methods, starting with `sendMessage`. First, import OpenZeppelin's `IERC20` contract, then in `sendMessage` check whether `feeAmount` is greater than zero. If it is, transfer and approve the amount of IERC20 asset at `feeTokenAddress` to the Teleporter Messenger saved as a state variable. Relayer fees are an optional way to incentive relayers to deliver a Teleporter message to its destination. They are not strictly necessary, and may be omitted if a relayer is willing to relay messages with no fee, such as with a self-hosted relayer.
+
+```solidity
+// For non-zero fee amounts, transfer the fee into the control of this contract first, and then
+// allow the Teleporter contract to spend it.
+if (feeAmount > 0) {
+    IERC20 feeToken = IERC20(feeTokenAddress);
+    require(
+        feeToken.transferFrom(msg.sender, address(this), feeAmount),
+        "Failed to transfer fee amount"
+    );
+    require(
+        feeToken.approve(address(teleporterMessenger), feeAmount),
+        "Failed to approve fee amount"
+    );
+}
+```
```

Comment:
> suggestion feature isn't letting me add suggestions for multiple code blocks, so just pasting my suggestion for this section. Basically just a few fixed typos and some added context:

Now it's time to implement the methods, starting with `sendMessage`. First, add the for import OpenZeppelin's `IERC20` contract to the top of your contract. 

```solidity
import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
```

then in `sendMessage` check whether `feeAmount` is greater than zero. If it is, transfer and approve the amount of IERC20 asset at `feeTokenAddress` to the Teleporter Messenger saved as a state variable. 


```solidity
function sendMessage(
    bytes32 destinationChainID,
    address destinationAddress,
    address feeTokenAddress,
    uint256 feeAmount,
    uint256 requiredGasLimit,
    string calldata message
) external returns (uint256 messageID) {
    // For non-zero fee amounts, first transfer the fee into the control of this contract,
    // then allow the Teleporter contract to spend it.
    if (feeAmount > 0) {
        IERC20 feeToken = IERC20(feeTokenAddress);
        require(
            feeToken.transferFrom(msg.sender, address(this), feeAmount),
            "Failed to transfer fee amount"
        );
        require(
            feeToken.approve(address(teleporterMessenger), feeAmount),
            "Failed to approve fee amount"
        );
    }
}
```

Note: Relayer fees are an optional way to incentivize relayers to deliver a Teleporter message to its destination. They are not strictly necessary, and may be omitted if a relayer is willing to relay messages with no fee, such as with a self-hosted relayer.

Reply:
might conflict with https://github.com/ava-labs/teleporter/pull/152